agent/consul/rpc.go

```diff
@@ -105,7 +105,7 @@ func (s *Server) handleConn(conn net.Conn, isTLS bool) {
 	typ := pool.RPCType(buf[0])
 
 	// Enforce TLS if VerifyIncoming is set
-	if s.tlsConfigurator.VerifyIncomingRPC() && !isTLS && typ != pool.RPCTLS && typ != pool.RPCTLSInsecure {
+	if s.tlsConfigurator.VerifyIncomingRPC() && !isTLS && typ != pool.RPCTLS && typ != pool.RPCTLSInsecure && typ != pool.RPCGRPC {
```

Comment:
Hmm. I'm not really understanding why we'd need to skip verification for GRPC? Seems like another silent security downgrade...

Reply:
Yeah, this was a mistake to leave in, good catch